src/main/java/org/gaul/s3proxy/S3Exception.java

```diff
@@ -27,33 +27,33 @@
     private final S3ErrorCode error;
     private final Map<String, String> elements;
 
-    S3Exception(S3ErrorCode error) {
+    public S3Exception(S3ErrorCode error) {
```

Comment:
Why make these public visibility?  I don't think the requests need their own subpackage so you can move them into the main package and reduce visibility here.